pkg/engine/renderer/enhancer.go

```diff
@@ -40,38 +40,24 @@ func (k *DefaultEnhancer) Apply(templates map[string]string, metadata Metadata)
 			if err != nil {
 				return nil, err
 			}
+
 			objUnstructured := &unstructured.Unstructured{Object: unstructMap}
+			objUnstructured.SetNamespace(metadata.InstanceNamespace)
 
-			labels := objUnstructured.GetLabels()
-			if labels == nil {
-				labels = make(map[string]string)
+			if err = addLabels(objUnstructured.Object, metadata); err != nil {
```

Comment:
seems strange to pass the map into unstructured as `Object` than pull it out for `addLabels` and `addAnnotations` 
would it make sense to create the unstructured object after we are done manipulating the map?

```
			if err = addLabels(unstructMap, metadata); err != nil {
				return nil, fmt.Errorf("adding labels on parsed object: %v", err)
			}
			if err = addAnnotations(unstructMap, metadata); err != nil {
				return nil, fmt.Errorf("adding annotations on parsed object: %v", err)
			}

			objUnstructured := &unstructured.Unstructured{Object: unstructMap}
			objUnstructured.SetNamespace(metadata.InstanceNamespace)

			if err = setControllerReference(metadata.ResourcesOwner, objUnstructured, k.Scheme); err != nil {
				return nil, fmt.Errorf("setting controller reference on parsed object: %v", err)
			}

```

Reply:
if not... I think I would prefer the `addLabels` to take the unstructured object and pull out the object... that way that function is more exact.

Reply:
Yeah. Makes sense, changed it.